### ISSUE TITLE
Option to lint on save

### ISSUE THREAD

**Issue by rafaelrenanpacheco** [redhat-developer/vscode-java#1355]
##### Environment
- Operating System: Linux x64 5.4.28-1-MANJARO
- JDK version: OpenJDK 11.0.6
- Visual Studio Code version: 1.43.2
- Java extension version: 0.59.0

##### Feature request
When typing some code the extension will automatically lint the code when I stop typing, but before I save the file. This causes some distraction because I get some warning even tho I didn't finish writing my code. Would be possible to add an option to lint the code on save in order to reduce linting distractions?

##### Current Result

Lint errors/warning while typing code, before any save is done.

##### Expected Result

Lint when saving the file.

**Comment by ok-nick** [comment]
This is also extremely useful for lower-end computers since it lags like hell as you type.

**Comment by jbytes1027** [comment]
I would also love this option!

**Comment by Pankaj-Talesara** [comment]
I would also love this option. Please make it happen

**Comment by isaacherbst613** [comment]
it would also be very useful to just have an ignore feature to ignore some errors

**Comment by Ronzeloo** [comment]
Yes it can be quite distracting when you're still typing the code and the extension already starts to report errors. An option to increase the error reporting delay from 1000ms to a higher value would be highly appreciated.
